lsp/src/Language/LSP/VFS.hs

```diff
@@ -458,15 +427,12 @@ codeUnitOffsetToCodePointOffset rope offset = do
  the position.
 -}
 codePointPositionToPosition :: VirtualFile -> CodePointPosition -> Maybe J.Position
-codePointPositionToPosition vFile (CodePointPosition l cpc) = do
+codePointPositionToPosition vFile (CodePointPosition l c) = do
   -- See Note [Converting between code points and code units]
   let text = _file_text vFile
-  utf16Line <- extractLine text (fromIntegral l)
-  -- Convert the line a rope using *code points*
-  let utfLine = URope.fromText $ Rope.toText utf16Line
-
-  cuc <- codePointOffsetToCodeUnitOffset utfLine (fromIntegral cpc)
-  pure $ J.Position l (fromIntegral cuc)
+  lineRope <- extractLine text $ fromIntegral l
+  guard $ c <= fromIntegral (Rope.charLength lineRope)
```

Comment:
I think we should include a comment about why we're not using `Rope.splitAtPosition`. The difference is quite subtle!

Reply:
splitAtPosition just wrapped over the line. when column index out of range or when the column indexing at the newline char :(

Reply:
The first out of range problem is easy to deal with,
when the column indexing at the newline char and still wrapped over the line, it becomes the pain.

Reply:
The first out of range problem is easy to deal with,
But the second case when the column indexing at the newline char and still wrapped over the line, it becomes the pain.

Reply:
comment added